### PR TITLE
Updated logic for creation of cgroup

### PR DESCRIPTION
**Pull request by diptanu**

### CODE REVIEW

File: client/driver/executor/executor_linux.go [hunk 1]
```diff
@@ -191,6 +196,9 @@ func DestroyCgroup(groups *cgroupConfig.Cgroup) error {
 	manager := getCgroupManager(groups)
 	if pids, perr := manager.GetPids(); perr == nil {
 		for _, pid := range pids {
+			if pid == os.Getpid() {
```
Comment (dadgar) [inline client/driver/executor/executor_linux.go]: Put a comment on why we do this

File: client/driver/executor/executor_test.go [hunk 2]
```diff
@@ -149,6 +149,50 @@ func TestExecutor_IsolationAndConstraints(t *testing.T) {
...
+
+	executor := NewExecutor(log.New(os.Stdout, "", log.LstdFlags))
+	ps, err := executor.LaunchCmd(&execCmd, ctx)
+	if err != nil {
+		t.Fatalf("error in launching command: %v", err)
+	}
+	if ps.Pid == 0 {
+		t.Fatalf("expected process to start and have non zero pid")
+	}
+	go func() {
```
Comment (dadgar) [inline client/driver/executor/executor_test.go]: Is it even necessary to wait?

**Comment by dadgar** [comment]
LGTM

File: client/driver/executor/executor_test.go (hunk 2 above)
Comment (diptanu) [inline client/driver/executor/executor_test.go]: So the Wait is going to block, and we won't be able to call Exit on the executor.

**Pull request merged**